src/main/java/microsoft/exchange/webservices/data/core/ExchangeServiceBase.java

```diff
@@ -230,6 +208,9 @@ private void initializeHttpPoolingClient() {
   public void setMaximumPoolingConnections(int maximumPoolingConnections) {
     if (maximumPoolingConnections < 1)
       throw new IllegalArgumentException("maximumPoolingConnections must be 1 or greater");
+    if(httpPoolingClient!=null){
```

Comment:
requirements are defined in [Googles Java Guide](https://google.github.io/styleguide/javaguide.html) which has been discussed to be our project reference. This is also as well mentioned in the [CONTRIBUTING.md](https://github.com/OfficeDev/ews-java-api/blob/master/CONTRIBUTING.md#coding-conventions). Using checkstyle has as well been discussed but comes along with several problems:
* usability of checkstyle-plugins (for different IDEs) differ from the checkstyle-google-styleguide
* project contains multiple formattings that where not touched by IntelliJ for some reasons these would have to be handled manually, maybe there are also bugs in the `codeStyleSettings.xml` didnt had that much time to step in.
If you have any ideas on how to do this "at low cost" feel free to discuss at #4. You are pushing at open doors with this. But whether or not we have these automated checks every PR has to match these criterias.